### PR TITLE
Add upper cut on jet time to HLTJetTimingFilter

### PR DESCRIPTION
**Pull request by ssantpur**
#### PR description:

This PR allows for more flexibility in choosing the timing thresholds for the delayed jet and delayed E/gamma triggers, this does not effect any existing triggers or their rates. The modification is to enable triggering on the negative timing region that can target beam halo and satellite collisions, and allow for maximum timing to be specified in the positive timing side to enable some delayed jet triggers to be added to parking dataset. 

#### PR validation:

I ran over HLTPhysics using Run2022G dataset and saw no change in the rates.

### CODE REVIEW

**Comment by missirol** [comment]
@ssantpur

For `HLTJetTimingFilter`, I can see that an upper cut on jet-time is missing and could be added.

But, in general, I don't see why it's necessary to allow for 2 time windows in the same instance of the filter. Couldn't one simply have two instances of the filter (one per time window), and combine the resulting jet/egamma candidates later on if needed ?

**Comment by ssantpur** [comment]
@missirol 
The upper cut on jet-time is added as jetMaxTimeThresh_ on the positive timing side (with a default value of 12.5ns) and as jetMaxNegTimeThresh_ on the negative timing side (with a default value of -12.5ns). The reason being we would like to allow for a single filter to be used for all the delayed jet applications. This allows us for easier manipulations or implementations in the HLT paths downstream instead of two instances of filters for the negative side.

Please let me know if this is a problem and I can have two instances of the same filter instead.

**Comment by missirol** [comment]
I think it's worth exploring the option of having 2 instances of the filter, before we complicate the logic of the filters themselves.

Feel free to share the corresponding HLT configuration, so we can judge what is the cleaner option.

**Comment by ssantpur** [comment]
@missirol 
After some discussion for the trigger rates of the satellite triggers and the parking of delayed jet paths, our needs have changed where
1. The Egamma filter is unchanged from the standard version. I reverted back to the original version of the filter.
2. Only one instance of the Jet Timing Filter is needed but with an addition of a maximum timing threshold.
This change is simple and does not complicate the existing filter and eliminates the need to add a duplicate instance of the trigger.
Please review the change and let me know if this is acceptable modification

File: HLTrigger/JetMET/plugins/HLTJetTimingFilter.h [hunk 1]
```diff
@@ -54,6 +55,7 @@ HLTJetTimingFilter<T>::HLTJetTimingFilter(const edm::ParameterSet& iConfig)
           consumes<edm::ValueMap<float>>(iConfig.getParameter<edm::InputTag>("jetEcalEtForTiming"))},
       minJets_{iConfig.getParameter<unsigned int>("minJets")},
       jetTimeThresh_{iConfig.getParameter<double>("jetTimeThresh")},
```
Comment (missirol) [inline HLTrigger/JetMET/plugins/HLTJetTimingFilter.h:57]: A clearer name for this parameter would have been `jetMinTimeThresh` (was missed in #35724), especially after this PR.

On the other hand, backporting this renaming to `13_0_X` would introduce a non-backward compatible change (to the module's PSet) in a closed release, which I would avoid. (and doing the renaming only in 13_1_X, but not in 13_0_X, makes it harder for TSG to migrate menus from `13_0_X` to `13_1_X`)

Unless there are suggestions, we'll open a GitHub issue to improve the parameter names in the future (without renaming in this PR).

**Review by missirol — COMMENTED** [review]
@ssantpur , please squash this PR in a single commit , after addressing the review comments.

**Comment by missirol** [comment]
@ssantpur , please also improve the title of the PR, e.g. "Add upper cut on jet time to `HLTJetTimingFilter`".

**Comment by missirol** [comment]
@ssantpur , please do the following to squash this PR into a single commit:
```bash
cmsrel CMSSW_13_0_1
cd CMSSW_13_0_1/src
cmsenv
git cms-init
git cms-addpkg HLTrigger/JetMET
git checkout be24c2d6ab1e83e8ea53dfd32abb8a7ff17c19e7 HLTrigger/JetMET/plugins/HLTJetTimingFilter.h
git checkout -b TwoSidedTimingFilter
git commit -m 'Add upper cut on jet time to HLTJetTimingFilter'
git push my-cmssw TwoSidedTimingFilter -f
```

**Comment by ssantpur** [comment]
@missirol 
I updated the branch as instructed and I agree that jetTimeThresh can be moved to jetMinTimeThresh in the future. It is not done here to allow for other HLT paths to continue using the jet timing filter without needing any modifications for v1.0. We can revisit it in the future. Let me know if anything else needs to be done.

**Comment by missirol** [comment]
> I updated the branch as instructed

I doubt that. As you can see, now the branch contains unrelated changes. Please, follow https://github.com/cms-sw/cmssw/pull/41089#issuecomment-1482859376. Otherwise, close this PR, and open a new PR with a single commit.

**Comment by missirol** [comment]
> I doubt that. As you can see, now the branch contains unrelated changes. Please, follow https://github.com/cms-sw/cmssw/pull/41089#issuecomment-1482859376. Otherwise, close this PR, and open a new PR with a single commit.

Scratch this. There was a mistake in my previous comment. You can start from `CMSSW_13_1_0_pre2` (instead of `13_0_1`).

**Pull request closed without merging**